qiskit/primitives/containers/data_bin.py

```diff
@@ -40,7 +41,15 @@ class DataBin(metaclass=DataBinMeta):
     :class:`make_dataclass`.
     """
 
-    _RESTRICTED_NAMES = ("_RESTRICTED_NAMES", "_SHAPE", "_FIELDS", "_FIELD_TYPES")
+    _RESTRICTED_NAMES = {
```

Comment:
I modified it from tuple to set because it is used only for membership operation.